minecode_pipelines/pipes/__init__.py

```diff
@@ -26,3 +39,120 @@ def write_data_to_file(path, data):
     path.parent.mkdir(parents=True, exist_ok=True)
     with open(path, encoding="utf-8", mode="w") as f:
         f.write(saneyaml.dump(data))
+
+
+def write_json_file(path, data):
+    path.parent.mkdir(parents=True, exist_ok=True)
+    with open(path, mode="w", encoding="utf-8") as f:
+        json.dump(data, f, indent=4, ensure_ascii=False)
+
+
+def write_purls_to_repo(repo, package, updated_purls, push_commits=False):
+    """Write or update package purls in the repo and optionally commit/push changes."""
+    ppath = hashid.get_package_purls_yml_file_path(package)
+    git_stage_purls(updated_purls, repo, ppath)
+    if push_commits:
+        commit_and_push_changes(repo)
+
+
+def git_stage_purls(purls, repo, purls_file):
+    """Write package URLs to a file and stage it in the local Git repository."""
+    relative_purl_file_path = Path(purls_file)
+
+    write_to = Path(repo.working_dir) / relative_purl_file_path
+
+    write_data_to_file(path=write_to, data=purls)
+
+    repo.index.add([relative_purl_file_path])
+    return relative_purl_file_path
+
+
+def commit_and_push_changes(repo):
```

Comment:
@ziadhany 

This file has overlap with the updated federatedcode pipes in https://github.com/aboutcode-org/scancode.io/blob/c69e8a0492b4e3abfde2f66dcd3434c883cf2ba2/scanpipe/pipes/federatedcode.py


Use scio from the commit https://github.com/aboutcode-org/scancode.io/commit/c69e8a0492b4e3abfde2f66dcd3434c883cf2ba2 and see if you can replace some of the code here with the functions from `scanpipe.pipes.federatedcode`